packages/vnf-wallet-sdk-nodejs/src/impl/data/verifiers/CredentialManifestByDeepLinkVerifierImpl.ts

```diff
@@ -10,19 +10,53 @@ import CredentialManifestByDeepLinkVerifier from '../../domain/verifiers/Credent
 import VCLLog from '../../utils/VCLLog';
 import VCLError from '../../../api/entities/error/VCLError';
 import VCLErrorCode from '../../../api/entities/error/VCLErrorCode';
+import ResolveDidDocumentRepository from '../../domain/repositories/ResolveDidDocumentRepository';
+import VCLDidDocument from '../../../api/entities/VCLDidDocument';
 
 export default class CredentialManifestByDeepLinkVerifierImpl
     implements CredentialManifestByDeepLinkVerifier
 {
+    constructor(
+        private readonly resolveDidDocumentRepository: ResolveDidDocumentRepository
+    ) {}
+
     async verifyCredentialManifest(
         credentialManifest: VCLCredentialManifest,
         deepLink: VCLDeepLink
     ): Promise<boolean> {
-        if (credentialManifest.issuerId === deepLink.did) {
+        if (deepLink.did === null) {
+            await this.onError(`DID not found in deep link: ${deepLink.value}`);
+            return false;
+        }
+        const didDocument =
+            await this.resolveDidDocumentRepository.resolveDidDocument(
+                deepLink.did!
+            );
+        return this.verify(credentialManifest, didDocument);
+    }
+
+    private async verify(
```

Comment:
why are there two methods that sound like they are doing the same thing? just create one long method?
I would have expected `verify` to also include the code that "verifies the JWT". That also means looking up the keys on the DID Document (stop using `resolve-kid`)